File: unrelated/LowestPriorityStrategy.cpp
#include "LowestPriorityStrategy.h"

#include <memory>

#include "HighestPriorityStrategy.h"

int LowestPriorityStrategy::winner(int minPriorityIndex, int maxPriorityIndex)
{
	return minPriorityIndex;
}

int LowestPriorityStrategy::looser(int minPriorityIndex, int maxPriorityIndex)
{
	return maxPriorityIndex;
}

bool LowestPriorityStrategy::leftIsBetter(double a, double b)
{
	return a < b;
}

std::shared_ptr<WinnerStrategy> LowestPriorityStrategy::getInverse()
{
	return std::make_shared<HighestPriorityStrategy>();
}

File: TSP/code/_electoralMethods/electoralsystems/util/LowestPriorityStrategy.java
package _electoralMethods.electoralsystems.util;

public class LowestPriorityStrategy implements WinnerStrategy {
    public int winner(int minPriorityIndex, int maxPriorityIndex) {
        return minPriorityIndex;
    }

    public int looser(int minPriorityIndex, int maxPriorityIndex) {
        return maxPriorityIndex;
    }

    public boolean leftIsBetter(double a, double b) {
        return a < b;
    }

    public WinnerStrategy getInverse() {
        return new HighestPriorityStrategy();
    }
}